modules/promotions/assets/js/react/controls/promotion.js

```diff
@@ -0,0 +1,88 @@
+import App from '../app';
+import { default as ControlBaseDataView } from 'elementor-controls/base-data';
+import { createRoot } from 'react-dom/client';
+
+export default class extends ControlBaseDataView {
+	promotionInfoTip = null;
+
+	selectors = {
+		reactAnchor: '.e-promotion-react-wrapper',
+	};
+
+	ui() {
+		return {
+			animatedHeadlineButton: '[data-promotion].elementor-control-type-switcher',
```

Comment:
```suggestion
			switcher: '[data-promotion].elementor-control-type-switcher',
```
This can be renamed to be more generic, right?

Reply:
Done

Reply:
@louiswol94 sure, on it..